sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryWebFilter.java

```diff
@@ -22,41 +23,28 @@
 /** Manages {@link io.sentry.Scope} in Webflux request processing. */
 @ApiStatus.Experimental
 @Open
-public class SentryWebFilter implements WebFilter {
-  private final @NotNull IHub hub;
-  private final @NotNull SentryRequestResolver sentryRequestResolver;
-  public static final String SENTRY_HUB_KEY = "sentry-hub";
+public class SentryWebFilter extends AbstractSentryWebFilter {
 
   public SentryWebFilter(final @NotNull IHub hub) {
-    this.hub = Objects.requireNonNull(hub, "hub is required");
-    this.sentryRequestResolver = new SentryRequestResolver(hub);
+    super(hub);
   }
 
   @Override
   public Mono<Void> filter(
       final @NotNull ServerWebExchange serverWebExchange,
       final @NotNull WebFilterChain webFilterChain) {
+    @NotNull IHub requestHub = Sentry.cloneMainHub();
     return webFilterChain
         .filter(serverWebExchange)
         .doFinally(
             __ -> {
-              hub.popScope();
+              doFinally(requestHub);
+              Sentry.setCurrentHub(NoOpHub.getInstance());
```

Comment:
This only resets one of the ThreadLocals. In testing the thread were `.doFirst` is run still keeps the hub alive. Don't know of a way to clean the ThreadLocal (yet) but I have another idea i wanna explore so we only leak an id and reference the hub in a store instead. I wanna try that in a different PR.